x/launch/types/initial_genesis.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import "errors"
+
+// NewInitialGenesisDefault returns an initial genesis that is the default genesis
+func NewInitialGenesisDefault() *InitialGenesis {
+	var ig InitialGenesis
+
+	ig.Source = &InitialGenesis_DefaultGenesis{
+		DefaultGenesis: &DefaultInitialGenesis{},
+	}
+
+	return &ig
+}
+
+// NewInitialGenesisURL returns an initial genesis from a genesis URL
+func NewInitialGenesisURL(genesisURL GenesisURL) *InitialGenesis {
+	var ig InitialGenesis
+
+	ig.Source = &InitialGenesis_GenesisURL{
+		GenesisURL: &genesisURL,
+	}
+
+	return &ig
+}
+
+// GetType returns the type of the initial genesis
+func (ig InitialGenesis) GetType() (InitialGenesisType, error) {
+	switch ig.Source.(type) {
+	case *InitialGenesis_DefaultGenesis:
+		return InitialGenesisType_DEFAULT, nil
+	case *InitialGenesis_GenesisURL:
+		return InitialGenesisType_URL, nil
+	default:
+		return InitialGenesisType_DEFAULT, errors.New("unrecognized initial genesis type")
+	}
+}
+
+// GenesisURL returns the genesis URL of the initial genesis if it's its type
+func (ig InitialGenesis) GenesisURL() (gURL GenesisURL, err error) {
```

Comment:
Is this getter used anywhere?

Reply:
We can remove those functions if you don't use them in `starport` :)